### PR TITLE
pkg/trace/test/testsuite: add credit card obfuscation test for all endpoints

### PR DESCRIPTION
**Pull request by Barbayar**
### What does this PR do?
This PR addresses non-critical feedback on https://github.com/DataDog/datadog-agent/pull/9757.

### Motivation
Code improvement.

### Additional Notes
N/A

### Possible Drawbacks / Trade-offs
None

### Describe how to test/QA your changes
No need to test. This PR has no logical change.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/api/api.go [hunk 1]
```diff
@@ -427,12 +427,10 @@ func decodeTracerPayload(v Version, req *http.Request, ts *info.TagStats) (tp *p
 			return nil, false, err
 		}
 		var tracerPayload pb.TracerPayload
-		err = tracerPayload.Unmarshal(buf.Bytes())
+		_, err = tracerPayload.UnmarshalMsg(buf.Bytes())
```
Comment (Barbayar) [inline pkg/trace/api/api.go:430]: This was an unfortunate mistake. `Unmarshal()` is proto decoder, and `UnmarshalMsg()` is msgp decoder.

Comment (gbbr) [reply pkg/trace/api/api.go:430]: This is crazy. We need an integration test for this or something that would've caught the problem. Or, is your newly added test the one that did catch it?

File: pkg/trace/test/testsuite/cards_test.go [hunk 2]
```diff
@@ -22,12 +30,44 @@ func TestCreditCards(t *testing.T) {
 	}()
 
 	for _, tt := range []struct {
-		conf []byte
-		out  string
+		conf    []byte
+		out     string
+		version string
```
Comment (gbbr) [inline pkg/trace/test/testsuite/cards_test.go:35]: Can you use `api.Version` ?
```suggestion
		version api.Version
```

File: pkg/trace/test/testsuite/cards_test.go [hunk 3]
```diff
@@ -36,27 +76,109 @@ apm_config:
...
 			waitForTrace(t, &r, func(v pb.AgentPayload) {
-				payloadsEqual(t, p, v)
+				payloadsEqual(t, traces, v)
 				assert.Equal(t, v.TracerPayloads[0].Chunks[0].Spans[0].Meta["credit_card_number"], tt.out)
 			})
 		})
 	}
 }
+
+func generatePayload(version string, cardNumber string) (msgp.Marshaler, pb.Traces) {
```
Comment (gbbr) [inline pkg/trace/test/testsuite/cards_test.go:101]:
```suggestion
func generatePayload(version, cardNumber string) (msgp.Marshaler, pb.Traces) {
```

**Review by gbbr — APPROVED** [review]

File: pkg/trace/api/api.go (hunk 1 above)
Comment (Barbayar) [reply pkg/trace/api/api.go:430]: Newly added test caught it.

File: pkg/trace/test/testsuite/cards_test.go (hunk 3 above)
Comment (Barbayar) [reply pkg/trace/test/testsuite/cards_test.go:101]: I'll use `api.Version` instead of `string`, here.

**Review by gbbr — APPROVED** [review]
Thanks!

**Pull request merged**